aspnetcore/fundamentals/minimal-apis/aspnetcore-openapi.md

```diff
@@ -0,0 +1,322 @@
+
+### [Visual Studio Code](#tab/visual-studio-code)
+
+Run the following command from the **Integrated Terminal**:
+
+```dotnetcli
+dotnet add package Swashbuckle.AspNetCore.SwaggerUi -v 6.5.0
+```
+
+### [.NET Core CLI](#tab/netcore-cli)
+
+Run the following command:
+
+```dotnetcli
+dotnet add package Swashbuckle.AspNetCore.SwaggerUi -v 6.5.0
+```
+
+---
+
+Then, enable the swagger-ui middleware with a reference to the OpenAPI route registered earlier. To limit information disclosure and security vulnerability concerns, it's recommend to only enable Swagger UI in development environments.
+
+[!code-csharp[](~/fundamentals/minimal-apis/9.0-samples/WebMinOpenApi/Program.cs?name=snippet_swaggerui)]
+
+### Linting generated OpenAPI documents with Spectral
+
+Spectral is an open-source OpenAPI document linter. Spectral can be incorporated into your application build to verify the quality of generated OpenAPI documents. Install Spectral according to the [package installation directions]https://github.com/stoplightio/spectral#-installation).
+
+To take advantage of Spectral, you must enable build-time OpenAPI document generation. To do so, install the `Microsoft.Extensions.ApiDescription.Server` package to enable build-time OpenAPI document generation.
+
+### [Visual Studio](#tab/visual-studio)
+
+* From the **Package Manager Console** window:
+  * Go to **View** > **Other Windows** > **Package Manager Console**
+  * Navigate to the directory in which the `.csproj` file exists
+  * Execute the following command:
+
+    ```powershell
+    Install-Package Microsoft.Extensions.ApiDescription.Server -v 9.0.0
+    ```
+
+* From the **Manage NuGet Packages** dialog:
+  * Right-click the project in **Solution Explorer** > **Manage NuGet Packages**
+  * Set the **Package source** to "nuget.org"
+  * Ensure the "Include prerelease" option is enabled
+  * Enter "Microsoft.Extensions.ApiDescription.Server" in the search box
+  * Select the latest "Microsoft.Extensions.ApiDescription.Server" package from the **Browse** tab and click **Install**
+
+### [Visual Studio Code](#tab/visual-studio-code)
+
+Run the following command from the **Integrated Terminal**:
+
+```dotnetcli
+dotnet add package Microsoft.Extensions.ApiDescription.Server -v 9.0.0
+```
+
+### [.NET Core CLI](#tab/netcore-cli)
+
+Run the following command:
+
+```dotnetcli
+dotnet add package Microsoft.Extensions.ApiDescription.Server -v 9.0.0
+```
+
+---
+
+Then, enable document generation at build-time by setting the following properties in your applications `.csproj` file.
+
+```xml
+<PropertyGroup>
+    <OpenApiDocumentsDirectory>$(MSBuildProjectDirectory)</OpenApiDocumentsDirectory>
+    <OpenApiGenerateDocuments>true</OpenApiGenerateDocuments>
+</PropertyGroup>
+```
+
+Run `dotnet build` to generate the document.
+
+```dotnetcli
+dotnet build
+```
+
+Create a `.spectral.yml` file with the following contents.
+
+```text
+extends: ["spectral:oas", "spectral:asyncapi"]
```

Comment:
Why include `asyncapi`? I don't think it would ever be used in this scenario.

Reply:
This is a result of me being a little lazy and copying the defaults from Spectral's docs 😅 . I'll update.